### PR TITLE
Fix next, setnext, cycle and recycle

### PR DESCRIPTION
**Pull request by Pablete1234**

### CODE REVIEW

**Comment by twizmwazin** [comment]
According to this commit it does 5 different things. To make it easier to review, can you break this into 5 commits?

File: src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java [hunk 1]
```diff
@@ -46,15 +50,13 @@
   private ClickEvent clickEvent;
   private HoverEvent hoverEvent;
   private List<BaseComponent> extra;
-  private boolean hover;
+  
+  private B thisObject;
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java]: `this` ?

File: src/main/java/in/twizmwaz/cardinal/component/map/MapComponent.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import lombok.Setter;
+import net.md_5.bungee.api.chat.BaseComponent;
+
+import java.util.Locale;
+
+@Getter
+@Setter
+public class MapComponent extends BaseLanguageComponent<MapComponent> {
+
+  LoadedMap map;
```
Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/component/map/MapComponent.java]: guess

File: src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java (hunk 1 above)
Comment (Pablete1234) [inline src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java]: it's a keyword, can't use that

Comment (twizmwazin) [inline src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java]: thisObject = this, right?

Comment (Pablete1234) [inline src/main/java/in/twizmwaz/cardinal/component/BaseComponentBuilder.java]: this is not an instance of "B"

**Comment by twizmwazin** [comment]
merged

**Pull request closed without merging**